.eslintrc.js

```diff
@@ -92,6 +92,10 @@ module.exports = {
 				selector: 'CallExpression[callee.name="withDispatch"] > :function > BlockStatement > :not(VariableDeclaration,ReturnStatement)',
 				message: 'withDispatch must return an object with consistent keys. Avoid performing logic in `mapDispatchToProps`.',
 			},
+			{
+				selector: 'LogicalExpression[operator="&&"][left.property.name="length"][right.type="JSXElement"]',
+				message: 'Avoid truthy checks on length property rendering, as zero length is rendered verbatim.',
```

Comment:
Will this still work?

https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

Reply:
> Will this still work?
> 
> https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

Do you mean the example of `unreadMessages.length > 0 &&` ? Yes, because in that case the left-hand of the LogicalExpression becomes a BinaryExpression (`.length > 0`), rather than the MemberExpression (`.length`) this rule is checking for.

You can also test it in the AST Explorer demo linked in the original comment:

https://astexplorer.net/#/gist/9ec7cf3fb53e63775defbc0e532b4cc8/487ccade54ed4c02fc846d1211083a9eac79f23f